### ISSUE TITLE
See if Ultralytics yolov3 Loss function can be implemented

### ISSUE THREAD

**Issue by ljbkusters** [ljbkusters/YOLOv3-1D#27]
I've seen a couple of loss functions (seemingly inspired by Ultralytics?) which implement the loss function slightly differntly. 

# Current Loss
In my current version I followed Alladin Pearssons tutorial where the loss is given by

$$\mathcal{L} = \sum_i \lambda_i \mathcal{L_i},\quad i \in \\{ \mathrm{obj}, \mathrm{noobj}, \mathrm{class}, \mathrm{box} \\}$$

Specifically these losses are computed as follows:

$$\mathcal{L}_\mathrm{obj}=\mathrm{BCEWithLogits}(o_p[\verb|obj_mask|], \mathrm{IOU}\cdot o_t[\verb|obj_mask|])$$

$$\mathcal{L}_\mathrm{noobj}=\mathrm{BCEWithLogits}(o_p[\verb|noobj_mask|], o_t[\verb|noobj_mask|])$$

$$\mathcal{L}_\mathrm{class}=\mathrm{CE}(c_p, c_t)\quad\text{with}\quad c_p, c_t\, \text{ onehot}$$

$$\mathcal{L}_\mathrm{box}=\mathrm{MSE}(\vec x_p, \vec x_t)\quad\text{with}\quad \vec x_i = (x_i, w_i)$$

with 

+ $\mathrm{BCEWithLogits}$ sigmoid followed binary cross entropy
+ $\mathrm{CE}$ Multi-class cross entropy
+ $\mathrm{MSE}$ Mean squared error
+ $(\cdot)_p$ predictions and $(\cdot)_t$ target
+ $\verb|obj_mask|$ equal to `argwhere(o_t == 1)`, $\verb|noobj_mask|$ equal to `argwhere(o_t == 0)`

# Ultralytics Loss
$\mathcal{L}_\mathrm{box}= (1.0-iou).mean()$
$\mathcal{L}_\mathrm{obj}= BCE()$
$\mathcal{L}_\mathrm{class}=BCE(c_p, c_t)$

**Comment by ljbkusters** [comment]
This no longer seems necessary. The bug described in issue #42 seems to have been te root cause of bad performance. Although slight differences in performance between the ultralytics loss fucntion and the current loss function may persist, they seem fairly irrelevant now that the model seems to be performing well. For this reason I'm closing this issue.
